modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryUtils.java

```diff
@@ -1998,6 +2027,14 @@ public static Object doReadOptimized(BinaryInputStream in, BinaryContext ctx, @N
         }
     }
 
+    /**
+     * @param ctx Context.
+     * @param clsLdr Class loader.
+     */
+    public static boolean isClassCacheWillUsed(BinaryContext ctx, ClassLoader clsLdr) {
```

Comment:
This method looks suspicion, moreover we have GridBinaryMarshaller.USE_CACHE.
Why we cannot use the thread local variable everywhere?